script/server

```diff
@@ -61,6 +61,22 @@ then
   fatal "config/diaspora.yml is missing! Copy over config/diaspora.yml.example to config/diaspora.yml and edit it properly!"
 fi
 
+# Check if detached head state
+git status | grep 'Changes not staged for commit' &> /dev/null
+if [ $? ]
+then
+  warning "Your changes are not staged for commit. Kindly stash or commit them"
+fi
+
+# Check if bundle is complete
+bundle check | grep 'Install missing gems' > /dev/null
```

Comment:
> Description:
>  Check searches the local machine for each of the gems requested in the Gemfile. If all gems are found, Bundler prints a success message and exits with a status of 0. If 
  not, the first missing gem is listed and Bundler exits status 1.

There's no need for the grep, just checking the return code is enough.